distributed/asyncio.py

```diff
@@ -166,13 +149,7 @@ def __del__(self):
             await to_asyncio_future(future)
             self.cluster.status = 'closed'
 
-        if self._make_current:
-            IOLoop.clear_current()
-
-    async def run_coroutine(self, function, *args, **kwargs):
-        tornfn = to_tornado(function)
-        future = self._run_coroutine(tornfn, *args, **kwargs)
-        return await to_asyncio_future(future)
+        BaseAsyncIOLoop.clear_current()
```

Comment:
Is it worth having a `try...finally` block so that `clear_current` is always called?